src/main/java/com/zendesk/maxwell/monitoring/MaxwellMetrics.java

```diff
@@ -89,17 +93,31 @@ private void setup(MaxwellConfig config) {
 						.build();
 			}
 
-			final DatadogReporter reporter = DatadogReporter.forRegistry(config.metricRegistry)
+			datadogReporter = DatadogReporter.forRegistry(config.metricRegistry)
 					.withTransport(transport)
 					.withExpansions(EnumSet.of(COUNT, RATE_1_MINUTE, RATE_15_MINUTE, MEDIAN, P95, P99))
 					.withTags(getDatadogTags(config.metricsDatadogTags))
 					.build();
 
-			reporter.start(config.metricsDatadogInterval, TimeUnit.SECONDS);
+			datadogReporter.start(config.metricsDatadogInterval, TimeUnit.SECONDS);
 			LOGGER.info("Datadog reporting enabled");
 		}
 	}
 
+	public void stop() {
+		if(slf4jReporter != null) {
+			slf4jReporter.stop();
+		}
+
+		if(jmxReporter != null) {
+			jmxReporter.stop();
+		}
+
+		if(datadogReporter != null) {
+			datadogReporter.stop();
+		}
+	}
+
```

Comment:
Is it safe to call `stop` multiple times on these? Or should you set each of these to `null` after stopping?

Reply:
good catch! I should have marked as null for those variables.